src/helpers/internal_viewport.ts

```diff
@@ -72,23 +66,29 @@ export class InternalViewport {
       this.sheetId,
       lastRow
     );
-    const leftColIndex = this.searchHeaderIndex("COL", lastColEnd - this.width, 0);
+
+    const leftColIndex = this.searchHeaderIndex("COL", lastColEnd - this.clientWidth, 0);
     const leftColSize = this.getters.getColSize(this.sheetId, leftColIndex);
-    const leftRowIndex = this.searchHeaderIndex("ROW", lastRowEnd - this.height, 0);
+    const leftRowIndex = this.searchHeaderIndex("ROW", lastRowEnd - this.clientHeight, 0);
     const topRowSize = this.getters.getRowSize(this.sheetId, leftRowIndex);
 
-    const width =
-      lastColEnd -
-      this.offsetCorrectionX +
-      (this.canScrollHorizontally
-        ? Math.max(DEFAULT_CELL_WIDTH, Math.min(leftColSize, this.width - lastColSize))
-        : 0);
-    const height =
-      lastRowEnd -
-      this.offsetCorrectionY +
-      (this.canScrollVertically
-        ? Math.max(DEFAULT_CELL_HEIGHT + 5, Math.min(topRowSize, this.height - lastRowSize))
-        : 0);
+    let width = lastColEnd - this.offsetCorrectionX;
+    if (this.canScrollHorizontally) {
+      width += Math.max(
+        DEFAULT_CELL_WIDTH, // leave some minimal space to let the user know they scrolled all the way
+        Math.min(leftColSize, this.clientWidth - lastColSize), // Add pixels that allows the snapping at maximum horizontal scroll
+        this.clientWidth - width // if the viewsport grid size is smaller than its client width, return client width
```

Comment:
In this case, it just results in `this.clientWidth`, right ? (+width - width cancel out)
I think I'd just to `Math.max(width, this.clientWidth)` at the end to try simplify this already not so easy max(min).
The new formatting with comment is definitely better.

Reply:
agreed, I was a bit too eager to save some lines ^^